holopy/core/tests/test_metadata.py

```diff
@@ -0,0 +1,666 @@
+        self.assertEqual(points.z.size, npts)
+
+    @attr("fast")
+    def test_z_defaults_to_zero_when_xy_passed(self):
+        np.random.seed(70)
+        npts = 21
+        x = np.random.randn(npts)
+        y = np.random.randn(npts)
+        points = detector_points(x=x, y=y)
+
+        self.assertTrue(np.all(points.z == 0))
+
+    # r, theta, phi tests:
+
+    @attr("fast")
+    def test_stores_rthetaphi_as_correct_shape_when_rthetaphi_passed(self):
+        np.random.seed(70)
+        npts = 21
+        r = np.random.randn(npts)
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+        self.assertEqual(points.r.size, npts)
+        self.assertEqual(points.theta.size, npts)
+        self.assertEqual(points.phi.size, npts)
+
+    @attr("fast")
+    def test_stores_rthetaphi_correct_values_when_rthetaphi_passed(self):
+        np.random.seed(70)
+        npts = 21
+        r = np.random.randn(npts)
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+        self.assertTrue(np.all(points.r == r))
+        self.assertTrue(np.all(points.theta == theta))
+        self.assertTrue(np.all(points.phi == phi))
+
+    @attr("fast")
+    def test_stores_r_as_array_when_scalar_r_passed(self):
+        np.random.seed(70)
+        npts = 21
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+        # Then we pick a scalar r:
+        r = np.random.randn(1).squeeze()
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+
+        self.assertTrue(np.all(points.r == r))
+        self.assertEqual(points.r.size, npts)
+
+    @attr("fast")
+    def test_r_defaults_to_inf_when_thetaphi_passed(self):
+        np.random.seed(70)
+        npts = 21
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(theta=theta, phi=phi)
+
+        self.assertTrue(np.all(np.isinf(points.r)))
+        self.assertEqual(points.r.size, npts)
+
+    # Other tests:
+    @attr("fast")
+    def test_data_is_stored_as_zeros_of_corect_size(self):
+        npts = 23
+        x, y, z = np.random.randn(3, npts)
+        points = detector_points(x=x, y=y, z=z)
+        self.assertEqual(points.size, npts)
+        self.assertTrue(np.all(points.values == 0))
+
+    @attr("fast")
+    def test_name_defaults_to_data(self):
+        x, y, z = np.random.randn(3, 10)
+        points = detector_points(x=x, y=y, z=z)
+        self.assertEqual(points.name, 'data')
+
+    @attr("fast")
+    def test_name_is_stored(self):
+        x, y, z = np.random.randn(3, 10)
+        name = 'this-is-a-test'
+        points = detector_points(x=x, y=y, z=z, name=name)
+        self.assertEqual(points.name, name)
+
+    # FIXME no checks for normal default values
```

Comment:
2. I haven't add a check for the default values for the normals in ```detector_points```, mostly because it seems like from recent discussions the default values will change soon. Thoughts? I could add one that tests the current defaults for now if desired.

Reply:
Verdict is to ignore the defaults for now, since normals will get changed soon.

Reply:
Marking as resolved then.